docs/reference/tags.rst

```diff
@@ -0,0 +1,98 @@
+Tags
+======
+
+sonata.admin.template_registry
```

Comment:
I don't fully understand this doc.

In the first case you defined a `app.admin.post.template_registry` but you never use it.
If it's not well documented, it will never be used.

Instead of a technical documentation about the tag, it seems better to have a functional documentation about the feature related: how to override an admin template.
- First case, directly in the configuration: thanks to the `sonata.admin.template_registry` tag.
- Second case, with a re-usable template registry (and how to do this).

Reply:
In the first case you do not have to use template registry with admin. This service can be standalone registry. Service name is confusing, I will change it. 

IMO template registry should be move to `sonata-project/twig-extensions` with this `tags.rst` file and documentation dedicated for override admin template should be add.

WDYT? @sonata-project/contributors

Reply:
I still don't understand how to use a template registry, so I cannot say if it's better in sonataAdmin or twigExtensions.

But currently, it's the only blocking point for Sonata 4.0.
So if we can't solve the situation, we should just un-deprecate `setTemplate`, remove templateRegistry and delay this feature to the next major.